### PR TITLE
feat: avoid missing injection not found in edge Vue 2 edge case

### PR DESCRIPTION
**Pull request by XRenSiu**
issue #1650

### CODE REVIEW

**Comment by posva** [comment]
This shouldn’t be necessary. Do you have a boiled down repro?

**Comment by XRenSiu** [comment]
> This shouldn’t be necessary. Do you have a boiled down repro?

Reproduction: https://stackblitz.com/edit/vitejs-vite-t9sovg

**Comment by posva** [comment]
Thanks!

The issue comes from the fact that `provide` doesn't work in the component passed to `new Vue()`, it does work in the one passed as `render: h => h(App)`. It counts as using the store globally, so passing the pinia instance will be necessary:

```ts
const app = new Vue({
  pinia,
  render: () => h(App),
  setup() {
    const store = useFooStore(pinia);
    const countStore = useCountStore(pinia);
    console.log(store.bar);
    console.log(countStore.count);
  },
});
```

However I think I have an idea to a simpler fix for this

File: packages/pinia/src/store.ts [hunk 1]
```diff
@@ -882,11 +882,21 @@ export function defineStore(
...
+        }
+      } else {
+        return currentInstance && inject(piniaSymbol)
+      }
+    }
     pinia =
       // in test mode, ignore the argument provided as we can always retrieve a
       // pinia instance with getActivePinia()
       (__TEST__ && activePinia && activePinia._testing ? null : pinia) ||
-      (currentInstance && inject(piniaSymbol))
```
Comment (posva) [inline packages/pinia/src/store.ts:889]: Let's instead revert to the old version and give a default value to `inject(piniaSymbol, null)`. This should bypass the error and still work without false positives

**Review by posva — CHANGES_REQUESTED** [review]

**Review by posva — APPROVED** [review]

**Pull request merged**